File: 724-find-pivot-index/724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        
        
        
        
        leftsum = 0
        
        totalsum = sum(nums)
        
        for i in range(len(nums)):
            
            diff = totalsum - nums[i] - leftsum
            
            if diff == leftsum:
                return i
            
            leftsum += nums[i]
            
        return -1
            
            
        
        
        
        

File: 724-find-pivot-index/724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int lsum = 0;
        int totalsum = 0;
        for(int i=0;i<nums.length;i++)
            totalsum += nums[i];
        int diff;
        for(int i=0;i<nums.length;i++){
            diff = totalsum - nums[i] - lsum;
            if(diff==lsum){
                return i;
            }
            lsum += nums[i];
        }
        return -1;
    }
}